pages/index.js

```diff
@@ -72,7 +72,15 @@ export default function Home({
 
   useEffect(() => {
     const refreshData = () => {
-      router.replace(router.asPath);
+      // Strip hash from URL. The __raydiantWindowId hash is added by Raydiant to communicate
+      // with the Raydiant SDK for features such as Dynamic Duration. Unfortunately, this causes
+      // the browser to do a client-side navigation instead of a server-side navigation, which doesn't
+      // refresh the data. To work around this, we strip the hash from the URL before refreshing.
+      //
+      // If you need to use the Raydiant SDK with NextJS, we recommend using a library like swr or
+      // react query to handle data fetching.
+      const refreshUrl = router.asPath.split('#')[0];
```

Comment:
At the moment I don't think we do because this is somewhat specific to NextJS. NextJS doesn't really have a way to easily refresh the data by calling `getServerSideProps` again. 

The recommended approach is to use the router to reload the current page: https://github.com/vercel/next.js/discussions/37194

But this doesn't work when there's a hash parameter. 

For client-side apps that aren't using NextJS you wouldn't be using the router to refresh data, you'd be polling in `setInterval` or using something like react-query like I did in this example: https://github.com/mirainc/starbucks-menu-app/commit/436850c337730d4174b10cd396c177446c9c5ddb

Reply:
I'd like to think if there are other ways to facilitate communication with the Raydiant SDK without a query param to avoid this altogether.